clients/fides-js/src/fides-tcf.ts

```diff
@@ -184,11 +184,18 @@ const init = async (config: FidesConfig) => {
       config.options.debug,
       "Explicit fidesString detected. Proceeding to override all TCF preferences with given fidesString"
     );
-    cookie.fides_string = config.options.fidesString;
-    cookie.tcf_consent = transformFidesStringToCookieKeys(
+    const { cookieKeys, success } = transformFidesStringToCookieKeys(
       config.options.fidesString,
       config.options.debug
     );
+    if (!success) {
+      // Treat an unsuccessful transform as if no fidesString were passed in.
+      // eslint-disable-next-line no-param-reassign
+      config.options.fidesString = null;
```

Comment:
I think `eslint` is right here... modifying that argument feels confusing to me, I'd want to be able to _remember_ that there was an override string provided.

Why do we need to set it to `null` here, is it being re-read again later?

Reply:
yeah that's the route I went first, but it turned out it was being used [here](https://github.com/ethyca/fides/blob/main/clients/fides-js/src/lib/initialize.ts#L315) as well, which right now just checks if a `fidesString` exists, but really needs to also know that the string is valid. we could have it check if it's valid instead, but I wasn't sure if calling decode again would eat into the initialization time... thoughts?

Reply:
I'll see if I can refactor the other function that uses `fidesString` 👍